code/datums/status_effects/wound_effects.dm

```diff
@@ -88,6 +89,17 @@
 			owner.client.move_delay += slowdown_right * determined_mod
 		next_leg = left
 
+/// We need to make sure that we properly clear these refs if one of the owner's limbs gets deleted
+/datum/status_effect/limp/proc/on_limb_removed(datum/source, obj/item/bodypart/limb_lost, special, dismembered)
+	SIGNAL_HANDLER
+
+	if(limb_lost == left)
+		left = null
+	if(limb_lost == right)
+		right = null
+
+	update_limp()
```

Comment:
I feel like this needs some elaboration, wouldn't
```js
	left = C.get_bodypart(BODY_ZONE_L_LEG)
	right = C.get_bodypart(BODY_ZONE_R_LEG)
```
from update_limp you call afterwards just instantly reassign these values again? Or is this some weird async race condition? (that happened before)